case-studies/antipatterns/classic/app/pods/side-effects/controller.js

```diff
@@ -1,23 +1,26 @@
-import Controller from '@ember/controller';
-import { computed } from '@ember/object';
+import Controller from "@ember/controller";
+import { computed } from "@ember/object";
 
 export default Controller.extend({
-  firstName: 'John',
-  lastName: 'Doe',
+  firstName: "John",
+  lastName: "Doe",
 
   fullNameLength: null,
 
-  fullName: computed('firstName', 'lastName', function () {
-    const fullName = `${this.get('firstName')} ${this.get('lastName')}`;
-
-    this.set('fullNameLength', fullName.length);
-
+  fullName: computed("firstName", "lastName", function () {
+    const fullName = `${this.get("firstName")} ${this.get("lastName")}`;
     return fullName;
   }),
 
+  fullNameLength: computed("fullName", function () {
+    if (this.get("fullName")) {
+      return this.get("fullName.length");
+    } else return null;
```

Comment:
Good job providing a default value!

Note that you can use `||` or `??` instead of `if`:

```js
  fullNameLength: computed("fullName", function () {
    return this.get("fullName.length") || null;
  })
```

The `??` is a new syntax that is not available in the exercise app, but it should be available in the `sibp` frontend.

More info:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing